### ISSUE TITLE
Launch/Debug multiple configurations at once

### ISSUE THREAD

**Issue by edvinv** [Microsoft/vscode#1616]
In launch.json you are able to define multiple launch configuration but you can only select one to run with F5. Is there a way to launch two configuration with single Debug command? I want to start and debug server side node app and also use chrome debug extension for client side, all from same VSC instance.

thx

**Comment by bpasero** [comment]
@isidorn fyi

**Comment by isidorn** [comment]
Currently this is not supported. In order to do this you need to open two instances of vscode

**Comment by buzinas** [comment]
> In order to do this you need to open two instances of vscode

That was my first try, but when trying to open the same project folder in a new instance, VSCode simply focus on the already opened window, instead of opening the project in the new one.

My team really needs this, it's terrible to stop client debugging, and then attach server one, and then stop, and attach client again. In fact, we're using the browser itself for client debugging, and VSCode for Node debugging, only because of this lack of support.

I have opened https://github.com/Microsoft/vscode-chrome-debug/issues/140 before actually finding this thread.

Is this feature becoming available anytime soon?

**Comment by isidorn** [comment]
fyi @weinand

**Comment by weinand** [comment]
@buzinas I recommend to create a sub folder for the server or client part of your project. Then you can easily create another VS Code window where you can run another debugger. We are using this approach  and it works great.

**Comment by buzinas** [comment]
@weinand Yes, since debugging both in the same instance would be much harder to accomplish - removing the limitation would be good enough (and probably an easy thing to ship).

Since our codebase huge, and we have lots of build processes / continuous integration etc etc, it would be kind of complex to make a folder refactor as you're suggesting. But I'll make a clone of the repo and give a try, maybe it's simpler than it seems.

**Comment by weinand** [comment]
@buzinas as I said, the sub-folder can be empty. The only thing that VS Code will create in that folder is another launch.json file. So no folder refactor is necessary.

**Comment by buzinas** [comment]
@weinand Could you please show me an example of that?

**Comment by weinand** [comment]
@buzinas here we go: 
![2016-04-01 15-49-19](https://cloud.githubusercontent.com/assets/1898161/14208835/9bd0e764-f821-11e5-8c81-ab8eadc0a029.png)

- Client and server code are in the same project.
- `_dummyFolder` only exists to be able to open another VS Code window on it.
- launch.json in _dummyFolder has 'go up' one level to refer to the server code.

**Comment by buzinas** [comment]
@weinand  Oh, that's really cool. I'll give it a try! Thank you :)

**Comment by weinand** [comment]
@gregvanl I think my workaround from above would be a good FAQ item for the debugger section?

**Comment by buzinas** [comment]
@weinand Although it's launching both here now, the breakpoints don't work.

It seems that the new instance doesn't send debugging info to the master one. How do you debug the server in that example of your screenshot?

**Comment by weinand** [comment]
@buzinas yes, you are right, the VS Code on the _dummyFolder has no access to source, so it cannot set breakpoints.

Solution: don't make the _dummyFolder a sub-folder but make it a super-folder:

![2016-04-01 16-35-23](https://cloud.githubusercontent.com/assets/1898161/14210175/1a84e38e-f828-11e5-9e04-463cd2a913b2.png)

**Comment by buzinas** [comment]
That's kindda problematic for our project.

But thanks for your time and attention! :)

For the next projects I'll use this approach, and for the current ones, I'll be waiting till we can open the same project in two different instances. Could this be done with an extension? If yes, I would consider developing it myself.

**Comment by weinand** [comment]
@buzinas but every folder has a parent folder, right? So VS Code can always open it (even if you cannot freely choose the name).

No, an extension cannot fix the issue that a folder cannot be opened in two VS Code windows. (Most likely the problem is just a single if-statement that needs to be removed ;-)

**Comment by thorseye** [comment]
Any updates on the possibility to attach multiple debuggers from the same VS Code window @weinand ? We've got a project with around 40 node processes (and counting), and I really like the ability to debug through VS Code. It's quite hard to follow a chain of requests through multiple services via the debugger if I cannot attach multiple debuggers at once. This was possible via multiple chrome tabs when using `node-inspector`.

**Comment by weinand** [comment]
This will be addressed as part of https://github.com/Microsoft/vscode/issues/13426

**Comment by ramya-rao-a** [comment]
Since this is being addressed as part of #13426, tagging this one as duplicate to help in our iteration tracking queries
